### PR TITLE
chore(deps): GHA updates

### PR DESCRIPTION
**Pull request by henryiii**
We are getting warnings about things like node 12 being removed soon.

### CODE REVIEW

**Review by thewtex — APPROVED** [review]

**Comment by henryiii** [comment]
I'm going to add a "pass" check, which will make this easier in the future. Currently we have to change the required checks every time a check gets renamed.

**Comment by mayeut** [comment]
> I'm going to add a "pass" check, which will make this easier in the future. Currently we have to change the required checks every time a check gets renamed.

Can't we just require `check_dist` to pass, it should be enough ? It already needs all the jobs needed by the new `pass` check.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -208,9 +209,15 @@ jobs:
...
+  pass:
+    needs: [build_wheels, build_manylinux2010_wheels, build_sdist, test_sdist, check_dist]
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "All jobs passed"
+
   upload_pypi:
     name: Upload to PyPI
-    needs: [build_wheels, build_manylinux2010_wheels, build_sdist, test_sdist, check_dist]
+    needs: [pass]
```
Comment (mayeut) [inline .github/workflows/build.yml:220]:
```suggestion
    needs: [check_dist]
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -208,9 +209,15 @@ jobs:
 
       - run: pipx run twine check --strict dist/*
 
+  pass:
+    needs: [check_dist]
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "All jobs passed"
+
```
Comment (henryiii) [inline .github/workflows/build.yml:217]:
```suggestion
```

I like the consistency of having a pass, but it is identical to check_dist, so okay, just use that for now. Similar to ninja.

**Pull request merged**